src/darker/formatters/formatter_config.py

```diff
@@ -0,0 +1,73 @@
+"""Code re-formatter plugin configuration type definitions."""
+
+from __future__ import annotations
+
+import re
+from decimal import Decimal
+from typing import TYPE_CHECKING, Iterable, Pattern, TypedDict
+
+from darkgraylib.config import ConfigurationError
+
+if TYPE_CHECKING:
+    from argparse import Namespace
+
+
+class FormatterConfig(TypedDict):
+    """Base class for code re-formatter configuration."""
+
+
+def validate_target_versions(
+    value: str | set[str], valid_target_versions: Iterable[str]
+) -> set[str]:
+    """Validate the target-version configuration option value."""
+    target_versions_in = {value} if isinstance(value, str) else value
+    if not isinstance(value, (str, set)):
+        message = f"Invalid target version(s) {value!r}"  # type: ignore[unreachable]
+        raise ConfigurationError(message)
+    bad_target_versions = target_versions_in - set(valid_target_versions)
+    if bad_target_versions:
+        message = f"Invalid target version(s) {bad_target_versions}"
+        raise ConfigurationError(message)
+    return target_versions_in
+
+
+get_version_num = re.compile(r"\d+").search
```

Comment:
Move below line 37

Reply:
I'm actually reworking the target version number handling, so this will go away altogether.